editor/filesystem_dock.cpp

```diff
@@ -1812,11 +1812,14 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 				fpath = p_selected[0];
 			}
 
-			if (!fpath.ends_with("/")) {
-				fpath = fpath.get_base_dir();
-			}
 			String dir = ProjectSettings::get_singleton()->globalize_path(fpath);
-			OS::get_singleton()->shell_open(String("file://") + dir);
+			if (fpath == "res://") {
+				// This case means that right click at the spcace area in FilesystemDock.
+				// Enter resource folder directly.
+				OS::get_singleton()->shell_show_in_explorer(dir, true);
+			} else {
+				OS::get_singleton()->shell_show_in_explorer(dir, false);
+			}
```

Comment:
Note that in such case the menu entry is "Open in File Manager", it's expected to open the folder instead of just showing it:
https://github.com/godotengine/godot/blob/0e7c98061856a171838f4ae41e2ca925158e24aa/editor/filesystem_dock.cpp#L2604

I don't think it needs to be changed.

```suggestion
			OS::get_singleton()->shell_show_in_explorer(dir, true);
```